OpenChange/MAPIStoreMailMessage.m

```diff
@@ -1394,6 +1394,28 @@ - (int) getPidLidMeetingType: (void **) data
           : MAPISTORE_ERR_NOT_FOUND);
 }
 
+- (int) getPidTagTransportMessageHeaders: (void **) data
+                    inMemCtx: (TALLOC_CTX *) memCtx
+{
+  NSDictionary *allHeaders = [sogoObject mailHeaders];
+  NSString *headersString = @"";
+  NSString *format= @"%@: %@\n";
+  int lastHeader = [allHeaders count] - 1;
```

Comment:
`[NSArray:count]` returns an `NSUInteger` as states in the documentation. Please, keep accurate to the typing.

https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Classes/NSArray_Class/index.html#//apple_ref/occ/instp/NSArray/count

Btw, I agree with @blaxter . Try to stick to the rich `NSString` library. For several reasons including well-tested and uniform along Objective-C code.